crates/core/tedge_agent/src/software_manager/actor.rs

```diff
@@ -101,7 +104,14 @@ impl Actor for SoftwareManagerActor {
 
         while let Some(request) = input_receiver.recv().await {
             tokio::select! {
-                _ = self.handle_request(request, &mut plugins, &operation_logs) => {}
+                res = self.handle_request(request, &mut plugins, &operation_logs) => {
+                    if let Err(SoftwareManagerError::Shutdown) = res {
+                        info!("Received shutdown request while handling the operation. Exiting...");
+                        // Returning an error here instead of break to exit with a non-zero exit code
+                        // return Err(RuntimeError::Shutdown);
```

Comment:
Just returning from here is not sufficient as it only shuts down the `SoftwareManager` actor and not the other actors. The runtime needs to catch those errors and issue shutdown requests to all the other actors as well.